assets/src/components/routePicker.tsx

```diff
@@ -9,23 +11,39 @@ interface Props {
   selectedRouteIds: RouteId[]
 }
 
-const RoutePicker = ({ routes, selectedRouteIds }: Props) => (
-  <div className="m-route-picker">
-    <h2>Selected Routes</h2>
-    {selectedRouteIds.length === 0 ? (
-      "none"
-    ) : (
+const RoutePicker = ({ routes, selectedRouteIds }: Props) => {
+  const routeFilter = useRouteFilter()
+
+  const filterRoutes = (allRoutes: Route[]): Route[] => {
+    const { filterType, filterText } = routeFilter
+
+    switch (filterType) {
+      case "id":
+        return allRoutes.filter((route: Route) => route.id.includes(filterText))
```

Comment:
If you continue to support different types of filters, this `filterRoutes` function should be defined inside `useRouteFilter.ts`. (If you simplify it so there are no filter types, the simple call to `.filter()` is fine living here.)